pkg/networkservice/mechanisms/memif/client.go

```diff
@@ -78,7 +89,11 @@ func (m *memifClient) Request(ctx context.Context, request *networkservice.Netwo
 		return nil, err
 	}
 
-	if err := create(ctx, conn, m.vppConn, metadata.IsClient(m)); err != nil {
+	// if direct memif enabled save socket filename to metadata
+	_, ok := loadDirectMemifInfo(ctx)
+	if mechanism := memif.ToMechanism(conn.GetMechanism()); mechanism != nil && ok {
+		storeDirectMemifInfo(ctx, directMemifInfo{socketURL: mechanism.GetSocketFileURL()})
+	} else if err := create(ctx, conn, m.vppConn, metadata.IsClient(m)); err != nil {
```

Comment:
I suppose that the following changes will keep code more straight and clean.
```suggestion
		return conn, nil
	}
	
	if err := create(ctx, conn, m.vppConn, metadata.IsClient(m)); err != nil {
```

Reply:
ok, done